src/compiler/c.rs

```diff
@@ -544,12 +545,13 @@ impl pkg::ToolchainPackager for CToolchainPackager {
                 output.stdout.pop();
             }
 
-            // Create our PathBuf from the raw bytes, and return if absolute.
+            // Create our PathBuf from the raw bytes.  Assume that relative
+            // paths can be found via PATH.
             let path: PathBuf = OsString::from_vec(output.stdout).into();
             if path.is_absolute() {
                 Some(path)
             } else {
-                None
+                which(path).ok()
```

Comment:
gcc seems to just echo its `-print-prog-name` argument back to stdout if you don't pass a tool it recognizes, which `as` seems to not be. Are we worried about finding random/wrong assemblers by accident here? Maybe that's not an issue in practice. In any case, if this fixes something you care about go ahead and merge it.

Reply:
I'm not sure what GCC wants, exactly; the GCC 5 I have on my Ubuntu machine just prints `as`, while GCC 6 prints an absolute path to `as`.  I think the original issue was using another, newer GCC that *didn't* print absolute paths, and I don't know why the behavior changed (or if there are custom distribution patches installed).

There are obscure ways to change what GCC looks for here without modifying `PATH`, but I'm not worried about running into them.